internal/handlers/common/distinct.go

```diff
@@ -29,11 +29,11 @@ import (
 //
 //nolint:vet // for readability
 type DistinctParams struct {
-	DB         string          `ferretdb:"$db"`
-	Collection string          `ferretdb:"collection"`
-	Key        string          `ferretdb:"key"`
-	Filter     *types.Document `ferretdb:"query,opt"`
-	Comment    string          `ferretdb:"comment,opt"`
+	DB         string `ferretdb:"$db"`
+	Collection string `ferretdb:"collection"`
+	Key        string `ferretdb:"key"`
+	Filter     any    `ferretdb:"query,opt"`
```

Comment:
This shouldn't be any type. `commonparams.ExtractParams` validates that fields contain valid type. But by replacing `any`, it no longer does the validation of the field type, so array, int, bool etc are all valid now here :disappointed_relieved:.  This comment is important :exclamation: